x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/support/SecurityIndexManagerTests.java

```diff
@@ -309,6 +311,60 @@ private ClusterChangedEvent event(ClusterState clusterState) {
         return new ClusterChangedEvent("test-event", clusterState, EMPTY_CLUSTER_STATE);
     }
 
+    public void testStateChangeListenerIsCalledIfMigrationIsRequired() {
```

Comment:
It would be nice to add an assertion in this test that makes sure that the listener isn't called if the migration is running and the state is unchanged, since that protects the listener from being called on every cluster state change until the target migration version has been applied.

Reply:
Fair, I can add that in a followup PR (or leave it to you if you want to get one in today).

I don't think we want to hold up the fix waiting for the test (since the code does check for an in-progress migration)